src/uwtools/rocoto.py

```diff
@@ -184,38 +193,53 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param e: The parent element to add the new element to.
         :param config: Configuration data for this element.
         """
-        operands, operators, strequality = self._dependency_constants
+        operators, strequality = self._dependency_constants
```

Comment:
Since `<datadep>` and `<timedep>` each need special treatment, that left only `<taskdep>` as a member of `operands`, so I removed that category and also gave `<taskdep>` its own handler.